iac-extensions/docker/src/test/java/org/sonar/iac/docker/parser/grammar/DockerLexicalGrammarTest.java

```diff
@@ -83,8 +83,37 @@ void testKeyword() {
       .matches("CMD\\\nfoo") // no valid syntax but will be parsed
       .matches("CMD\\foo") // no valid syntax but will be parsed
       .notMatches("CMD[\"foo\"]")
+      .notMatches("CMD] src dest")
       .notMatches("CMDA foo")
       .notMatches("CMD8 foo")
-    ;
+      .notMatches("CMD-")
+      .notMatches("CMD_")
+      .notMatches("CMD<")
+      .notMatches("CMD>")
+      .notMatches("CMD!")
+      .notMatches("CMD@")
+      .notMatches("CMD#")
+      .notMatches("CMD$")
+      .notMatches("CMD% src dest")
+      .notMatches("CMD& src dest")
+      .notMatches("CMD£ src dest")
+      .notMatches("CMD§ src dest")
+      .notMatches("CMD` src dest")
+      .notMatches("CMD~ src dest")
+      .notMatches("CMD* src dest")
+      .notMatches("CMD( src dest")
+      .notMatches("CMD) src dest")
+      .notMatches("CMD= src dest")
+      .notMatches("CMD{ src dest")
```

Comment:
That's a lot of use cases XD I understand that you may want to test/ensure/prevent regression in the future, still that seems a lot, and from my perspective even with that big list of chars, there will still probably some other character.

I don't have a strong opinion on it, If I really wanted to have such restriction I probably would have written test differently, like a list of forbidden character and some loop / parameterized test.

Reply:
The reason for that test was that I realized that the old implementation of  `DockerLexicalGrammar.keywords()` had a bug and `CMD` + special character, was parsed in the wrong way. 
I will improve this test in a separate PR to unblock other work on the project.